src/test/kotlin/org/rust/ide/inspections/import/AutoImportFixTest.kt

```diff
@@ -549,7 +549,8 @@ class AutoImportFixTest : AutoImportFixTestBase() {
         }
     """)
 
-    fun `test don't import trait method`() = checkAutoImportFixIsUnavailable("""
+    fun `test don't import trait assoc function if its import is useless`() = expect<IllegalStateException> {
```

Comment:
Initially, I want to disallow such imports by this check:
```kotlin
/**
 * If function is defined in a trait
 * ```rust
 * trait Trait {
 *     fn foo() {}
 * }
 * ```
 * it potentially can be accessed by the trait name `Trait::foo` only if there are `self` parameter or
 * `Self` type in the function signature
 */
val RsFunction.canBeAccessedByTraitName: Boolean
    get() {
        check(owner is RsAbstractableOwner.Trait)
        return selfParameter != null || typeOfValue.visitWith(object : TypeVisitor {
            override fun visitTy(ty: Ty): Boolean =
                if (ty is TyTypeParameter && ty.parameter is TyTypeParameter.Self) true else ty.superVisitWith(this)
        })
    }
```

but looks like this solution doesn't work with default methods accessed by struct name (I can't be sure without #4070) and the case we want forbid is very rare (why use trait functions without `Self` types and why?) so I this we can allow it for now

Reply:
> I can't be sure without #4070

You can check it now

Reply:
> You can check it now

Checked, and just rewrote it to work with default methods